### PR TITLE
Update protocol for regions use cases

### PR DESCRIPTION
**Pull request by volodkevych**
## Problem

There is no way to request chat UI to open a new tab from server. This will be needed to restore chat state after profile switch.

Given connection type becomes required now for region detection, we should move it to `aws/credentails/token/update` request - it simplifies the flow and less error-prone.

## Solution

- extended chat protocol with `'aws/chat/openTab'`
- extended `UpdateCredentialsParams` in `'aws/credentials/token/update'` with `startUrl`
- moved the logic to define connection type to runtimes

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtimes/runtimes/auth/auth.ts [hunk 1]
```diff
@@ -137,6 +143,18 @@ export class Auth {
         })
     }
 
+    private async handleBearerCredentialsMetadata(metadata?: ConnectionMetadata) {
+        if (metadata) {
+            this.connectionMetadata = metadata
```
Comment (viktorsaws) [inline runtimes/runtimes/auth/auth.ts:148]: Guess would be good to do more validation of incoming `metadata` object, since client could send whatever. Also not sure if we should return an error to client here, but at least logging something would be good.

File: runtimes/runtimes/auth/auth.ts [hunk 2]
```diff
@@ -70,6 +72,11 @@ export class Auth {
             getConnectionMetadata: () => {
                 return this.connectionMetadata
             },
+
+            getConnectionType: () => {
+                const startUrl = this.connectionMetadata?.sso?.startUrl
+                return !startUrl ? 'none' : startUrl.includes(BUILDER_ID_START_URL) ? 'builderId' : 'identityCenter'
```
Comment (viktorsaws) [inline runtimes/runtimes/auth/auth.ts:78]: nit: missing unit test for when `getConnectionType` returns `none`

**Review by viktorsaws — APPROVED** [review]

File: runtimes/runtimes/auth/auth.ts (hunk 1 above)
Comment (volodkevych) [reply runtimes/runtimes/auth/auth.ts:148]: All the fields in `metadata` are optional. Given that I'm not sure I should validate any other fields. If client send the metadata, he defined it already, even if it's empty.

File: runtimes/runtimes/auth/auth.ts (hunk 2 above)
Comment (volodkevych) [reply runtimes/runtimes/auth/auth.ts:78]: added it

**Review by viktorsaws — APPROVED** [review]
